include/openenclave/internal/defs.h

```diff
@@ -71,4 +71,12 @@
 /* OE_FIELD_SIZE */
 #define OE_FIELD_SIZE(TYPE, FIELD) (sizeof(((TYPE*)0)->FIELD))
 
+/* OE_MODULE_INIT */
+#ifdef __GNUC__
+#define OE_MODULE_INIT __attribute__((used, constructor))
+#define OE_MODULE_INIT_PRIORITY(__p) __attribute__((used, constructor(__p)))
```

Comment:
@anakrish one though. If we ever support dynamic loading, this doesn't really work. Constructor priority wouldn't be guaranteed accross .so files.

The staged model (of having something like early, default, late initialization stages) is a semantic we could extend to statically linked or dynamically linked module initialization.